python/lsst/ip/isr/isrFunctions.py

```diff
@@ -546,10 +560,30 @@ def overscanCorrection(ampMaskedImage, overscanImage, fitType='MEDIAN', order=1,
         stdevBiasArr = 0.74*(percentiles[2] - percentiles[0])  # robust stdev
         diff = numpy.abs(biasArray - medianBiasArr[:, numpy.newaxis])
         biasMaskedArr = numpy.ma.masked_where(diff > numSigmaClip*stdevBiasArr[:, numpy.newaxis], biasArray)
-        collapsed = numpy.mean(biasMaskedArr, axis=1)
-        if collapsed.mask.sum() > 0:
-            collapsed.data[collapsed.mask] = numpy.mean(biasArray.data[collapsed.mask], axis=1)
+
+        if fitType in ('MEDIAN_PER_ROW',):
+            collapsed = []
+            fitTypeStats = afwMath.stringToStatisticsProperty('MEDIAN')
+            # Loop over rows because there is no afwMath.Median(array, axis=1)
+            for row in biasMaskedArr:
+                if row.mask is False:
+                    row.mask = numpy.repeat(False, len(row))
```

Comment:
And even then, I'm not quite sure what this is doing really.

Reply:
The second argument of `medianRow = afwMath.makeStatistics(row, row.mask, fitTypeStats, statControl).getValue()`---one line below---is expecting a list. It fails if it just `False`, so I'm creating a list of `False` to pass a valid argument.

Reply:
I don't think you have to pass in a mask at all. Can't you just remove the argument and call with a different signature?

Reply:
Also, as discussed OOB, I don't think we _ever_ want to mask a row - it makes sense for things where a spline or poly is being fitted up the column, but for a row-wise median we _have_ to have a number for every row.